kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterFactory.java

```diff
@@ -57,14 +60,14 @@ public interface FilterFactory<C, I> {
      * @param initializationData The initialization data that was returned from {@link #initialize(FilterFactoryContext, Object)}.
      * @return the Filter instance.
      */
-    Filter createFilter(FilterFactoryContext context, I initializationData);
+    Filter createFilter(FilterFactoryContext context, @UnknownNullness I initializationData);
 
     /**
      * Called by the runtime to release any resources associated with the given {@code initializationData}.
      * This is guaranteed to eventually be called for each successful call to {@link #initialize(FilterFactoryContext, Object)}.
      * Once this method has been called {@link #createFilter(FilterFactoryContext, Object)} won't be called again.
      * @param initializationData The initialization data that was returned from {@link #initialize(FilterFactoryContext, Object)}.
      */
-    default void close(I initializationData) {
+    default void close(@UnknownNullness I initializationData) {
```

Comment:
Is the distinction here that it varies by implementation so we can't document when/why it would be null?